### PR TITLE
[main] Changelog for privacy manifest and feature branch merge

### PR DESCRIPTION
**Pull request by KunJeongPark**
### Summary of changes

Added changelog entry for Privacy manifest and merge feature branch with privacy manifest files for all modules

### Checklist

- [x] Added a changelog entry

- @KunJeongPark

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Braintree iOS SDK Release Notes
 
+## 6.15.0 (2024-03-18)
```
Comment (jaxdesmarais) [inline CHANGELOG.md:3]:
```suggestion
## unreleased (CI will not run with this filled in)
```

Comment (jaxdesmarais) [reply CHANGELOG.md:3]: This is where CI greps for unreleased: https://github.com/braintree/braintree_ios/blob/8775b31e1e003df7a67bf1428b784eec0637364a/.github/workflows/release.yml#L21-L22. And here is will CI updates the version and date: https://github.com/braintree/braintree_ios/blob/8775b31e1e003df7a67bf1428b784eec0637364a/.github/workflows/release.yml#L38

Comment (KunJeongPark) [reply CHANGELOG.md]: addressed here: c5604acb88ab23f9ff537e5cbc54b4ac565a4bd6

**Review by agedd — APPROVED** [review]
🚀

**Pull request merged**